rbtcwrapperproxy/RBTCWrapperProxy.sol

```diff
@@ -227,4 +251,98 @@ contract RBTCWrapperProxy{
         }        
     }
 
+    function addLiquidityToV1(
+        address _liquidityPoolConverterAddress,
+        IERC20Token[] memory _reserveTokens, 
+        uint256[] memory _reserveAmounts, 
+        uint256 _minReturn
+    )  
+        public
+        payable
+        checkAddress(_liquidityPoolConverterAddress)
+        returns(uint256) 
+    {
+        uint256 lengthOfTokens = _reserveTokens.length;
+
+        require(_reserveAmounts.length == lengthOfTokens.add(1), "The length of array is wrong");
+        require(_reserveAmounts[lengthOfTokens] == msg.value, "The provided amount of RBTC should be identical to msg.value");
+
+        bool successOfTransfer;
+        bool successOfApprove;
+        
+        for (uint256 i = 0; i < lengthOfTokens; i++) {
+            IERC20Token reserveToken = _reserveTokens[i];
+            uint256 reserveAmount = _reserveAmounts[i];
+            successOfTransfer = IERC20Token(reserveToken).transferFrom(msg.sender, address(this), reserveAmount);
+            require(successOfTransfer, "Failed to transfer reserve token from user");
+            successOfApprove = IERC20Token(reserveToken).approve(_liquidityPoolConverterAddress, reserveAmount);
+            require(successOfApprove, "Failed to approve converter to transfer reserve token");
+        }
+
+        (bool successOfDeposit, ) = wrbtcTokenAddress.call.value(_reserveAmounts[lengthOfTokens])(abi.encodeWithSignature("deposit()"));
+        require(successOfDeposit, "Failed to deposit RBTC to get WRBTC");
+        _reserveTokens[lengthOfTokens] = IERC20Token(wrbtcTokenAddress);
```

Comment:
`_reserveTokens` is a `memory` array with `lengthOfTokens` length, therefore `_reserveTokens[lengthOfTokens]` should fail with assert-style error.
If it works, please show me where I'm wrong.